src/bin/conllx-select.rs

```diff
@@ -0,0 +1,55 @@
+extern crate conllx;
+extern crate conllx_utils;
+extern crate getopts;
+extern crate stdinout;
+
+use std::env::args;
+use std::fs;
+use std::io::BufWriter;
+
+use conllx::WriteSentence;
+use conllx_utils::or_exit;
+use getopts::Options;
+use stdinout::{Input, Output};
+
+fn print_usage(program: &str, opts: Options) {
+    let brief = format!(
+        "Usage: {} [options] ID_FILE [INPUT_FILE] [OUTPUT_FILE]",
+        program
+    );
+    print!("{}", opts.usage(&brief));
+}
+
+fn main() {
+    let args: Vec<String> = args().collect();
+    let program = args[0].clone();
+
+    let mut opts = Options::new();
+    opts.optflag("h", "help", "print this help menu");
+    let matches = or_exit(opts.parse(&args[1..]));
+
+    if matches.opt_present("h") {
+        print_usage(&program, opts);
+        return;
+    }
+
+    if matches.free.is_empty() || matches.free.len() > 3 {
+        print_usage(&program, opts);
+        return;
+    }
+
+    let ids = or_exit(fs::read_to_string(&matches.free[0]));
+    let ids = ids
+        .split_whitespace()
+        .map(|s| or_exit(s.parse::<usize>()))
+        .collect::<Vec<_>>();
+
+    let input = Input::from(matches.free.get(1));
+    let reader = conllx::Reader::new(or_exit(input.buf_read()));
+    let corpus = reader.into_iter().map(Result::unwrap).collect::<Vec<_>>();
```

Comment:
`or_exit` in place of `unwrap`

Reply:
This could also be done without reading the whole corpus into memory. Put the ids in a set, go over the corpus with enumeration, writing if the ids match.

Reply:
That would complicate preserving the order given by the IDs since sentences would be printed in the order they are in the full treebank instead.

IIrc, I wrote that binary to grab the sentences corresponding to a train/dev/val split where changing order wouldn be unexpected.